source/databricks/settlement_report/settlement_report_job/domain/monthly_amounts/read_and_filter.py

```diff
@@ -0,0 +1,146 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from pyspark.sql import DataFrame
+from pyspark.sql.functions import lit, col
+
+from settlement_report_job import logging
+from settlement_report_job.domain.market_role import MarketRole
+from settlement_report_job.domain.repository import WholesaleRepository
+from settlement_report_job.wholesale.column_names import DataProductColumnNames
+from settlement_report_job.domain.settlement_report_args import SettlementReportArgs
+from settlement_report_job.domain.dataframe_utils.factory_filters import (
+    filter_by_calculation_id_by_grid_area,
+    filter_by_energy_supplier_ids,
+)
+
+log = logging.Logger(__name__)
+
+
+@logging.use_span()
+def read_and_filter_from_view(
+    args: SettlementReportArgs, repository: WholesaleRepository
+) -> DataFrame:
+    monthly_amounts_per_charge = repository.read_monthly_amounts_per_charge_v1()
+    monthly_amounts_per_charge = _filter_monthly_amounts_per_charge(
+        monthly_amounts_per_charge, args
+    )
+
+    total_monthly_amounts = repository.read_total_monthly_amounts_v1()
+    total_monthly_amounts = _filter_total_monthly_amounts(total_monthly_amounts, args)
+    total_monthly_amounts = _extend_total_monthly_amounts_columns_for_union(
+        total_monthly_amounts,
+        monthly_amounts_per_charge_column_ordering=monthly_amounts_per_charge.columns,
+    )
+
+    monthly_amounts = monthly_amounts_per_charge.union(total_monthly_amounts)
+    monthly_amounts = _extend_monthly_amounts_with_resolution(monthly_amounts)
+    monthly_amounts = _drop_columns_based_on_requester(
+        monthly_amounts, args.requesting_actor_market_role
+    )
+
+    return monthly_amounts
+
+
+def _apply_shared_filters(df: DataFrame, args: SettlementReportArgs) -> DataFrame:
+    df = df.where(
+        (col(DataProductColumnNames.time) >= args.period_start)
+        & (col(DataProductColumnNames.time) < args.period_end)
+    )
+
+    if args.calculation_id_by_grid_area:
+        # Can never be null, but mypy requires it be specified
+        df = df.where(
+            filter_by_calculation_id_by_grid_area(args.calculation_id_by_grid_area)
+        )
+
+    if args.energy_supplier_ids:
+        df = df.where(filter_by_energy_supplier_ids(args.energy_supplier_ids))
+
+    if args.requesting_actor_market_role in [
+        MarketRole.GRID_ACCESS_PROVIDER,
+        MarketRole.SYSTEM_OPERATOR,
+    ]:
+        df = df.where(
+            col(DataProductColumnNames.charge_owner_id) == args.requesting_actor_id
+        )
+
+    return df
+
+
+def _filter_monthly_amounts_per_charge(
+    monthly_amounts_per_charge: DataFrame, args: SettlementReportArgs
+) -> DataFrame:
+    monthly_amounts_per_charge = _apply_shared_filters(monthly_amounts_per_charge, args)
+
+    if args.requesting_actor_market_role == MarketRole.GRID_ACCESS_PROVIDER:
+        monthly_amounts_per_charge = monthly_amounts_per_charge.filter(
```

Comment:
After talking to Khatozen today, I think this is not what we want. Instead it should be 
```suggestion
        monthly_amounts_per_charge = monthly_amounts_per_charge.filter(
            col(DataProductColumnNames.is_tax) | col(DataProductColumnNames.charge_owner_id)==requesting_actor_id)
        )
```

Reply:
See also the PR for wholesale_resutls:
https://github.com/Energinet-DataHub/opengeh-wholesale/pull/2961/files

Reply:
I have taken their code, and added it + unit tests for their filter.